h/util/metrics.py

```diff
@@ -68,18 +32,24 @@ def record_search_api_usage_metrics(params):
         # Record usage of tags/tag (tags is an alias of tag).
         "tags",
         "tag",
-        # Record usage of _separate_replies which will help distinguish client calls
-        # for loading the sidebar annotations from other api calls.
-        "_separate_replies",
         # Record group and user-these help in identifying slow queries.
         "group",
         "user",
         # Record usage of wildcard feature.
         "wildcard_uri",
     ]
+    # The New Relic Query Language does not permit _ at the begining
+    # and offset is a reserved key word.
+    params = [("es_{}".format(k), params[k]) for k in keys if k in params]
 
-    for k in keys:
-        if k in params:
-            # The New Relic Query Language does not permit _ at the begining
-            # and offset is a reserved key word.
-            send_metric("es_{}".format(k), params[k])
+    # Record usage of _separate_replies which will help distinguish client calls
+    # for loading the sidebar annotations from other api calls.
+    if separate_replies:
+        params.append(("es__separate_replies", separate_replies))
+
+    # On startup, there is a race condition in NewRelic where there may not be
+    # a transaction. If there isn't, current_transaction will return None in which
```

Comment:
It's expected known behavior. We didn't have to worry about this before because we were calling newrelic.agent.add_custom_parameter in a loop and that function internally took care of whether the current transaction existed or not. I have filed a support ticket with them to expose this particular function (newrelic.agent.current_transaction.add_custom_parameters) on newrelic.agent in which case the current transaction handling would be done inside there as well but for now we have to handle it ourselves or take a slight performance hit by looping. Of course the performance hit is slight and I might be over-optimizing. I'm fine with either.